libnavui-maps/src/test/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtilsTest.kt

```diff
@@ -74,6 +74,30 @@ class MapboxRouteLineUtilsTest {
         assertEquals(expectedExpression, result.toString())
     }
 
+    @Test
+    fun getTrafficLineExpressionDuplicateOffsetsRemoved() {
+        val expectedExpression = "[step, [line-progress], [rgba, 0.0, 0.0, 0.0, 0.0], 0.0, " +
+            "[rgba, 86.0, 168.0, 251.0, 1.0], 0.7868200761181402, " +
+            "[rgba, 86.0, 168.0, 251.0, 1.0], 0.7930120224665551, " +
+            "[rgba, 86.0, 168.0, 251.0, 1.0], 0.7932530928525063, " +
+            "[rgba, 86.0, 168.0, 251.0, 1.0], 0.7964017663976524, [rgba, 86.0, 168.0, 251.0, 1.0]]"
+        val expressionDatas = listOf(
+            RouteLineExpressionData(0.7868200761181402, -11097861),
+            RouteLineExpressionData(0.7930120224665551, -11097861),
+            RouteLineExpressionData(0.7932530928525063, -11097861),
+            RouteLineExpressionData(0.7932530928525063, -11097861),
```

Comment:
What could cause duplicates in the `RouteLineExpressionData`?

Quickly browsing through the code, it seems like this could only happen if the running distance for 2 distinct points on the route is the same, which sounds like a clear bug:
https://github.com/mapbox/mapbox-navigation-android/blob/04d955826c5681c661e42742ca822847d52b1e62/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/internal/route/line/MapboxRouteLineUtils.kt#L292

I'm fine merging the workaround, but this isn't tackling the root cause. Either our parser is incorrect, or we're receiving invalid data from the Directions API.

@cafesilencio could you cut a follow-up ticket to investigate the root cause and possibly ask the customer for a route that reproduces the issue?

Reply:
The customer has been asked for the route but they can't reproduce the issue.